### PR TITLE
Introduce a smoke test for Compass director and ORD service

### PR DESCRIPTION
**Pull request by dzahariev**
**Description**

Changes proposed in this pull request:

- Introduce a smoke test for Compass director and ORD service

### CODE REVIEW

File: prow/jobs/incubator/compass/compass-ord.yaml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                cpu: 2
+        tolerations:
+          - key: dedicated
+            operator: Equal
+            value: high-cpu
+            effect: NoSchedule
+        nodeSelector:
+            dedicated: "high-cpu"
+  
+postsubmits: # runs on main
```
Comment (dzahariev) [inline prow/jobs/incubator/compass/compass-ord.yaml:56]: Remove post execution.

File: prow/scripts/provision-vm-compass-ord.sh [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+trap cleanup exit INT
+
+chmod -R 0777 /home/prow/go/src/github.com/kyma-incubator/compass/.git
+mkdir -p /home/prow/go/src/github.com/kyma-incubator/compass/components/console/shared/build
+
+log::info "Copying Compass to the instance"
+#shellcheck disable=SC2088
+utils::compress_send_to_vm "${ZONE}" "compass-ord-test-${RANDOM_ID}" "/home/prow/go/src/github.com/kyma-incubator/compass" "~/compass"
+
```
Comment (dzahariev) [inline prow/scripts/provision-vm-compass-ord.sh:113]: Get the version that is listed in values.yaml for ORD service and checkout the branch for the listed PR there.

**Review by NickyMateev — CHANGES_REQUESTED** [review]

File: prow/scripts/provision-vm-compass-ord.sh [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+log::info "ORD_PR_NUMBER PR is: ${ORD_PR_NUMBER}"
+
+ORD_PR_DATA=$(curl -sS "https://api.github.com/repos/kyma-incubator/ord-service/pulls/${ORD_PR_NUMBER}")
+log::info "ORD_PR_DATA is: ${ORD_PR_DATA}"
+
+ORD_PR_COMMIT_HASH=$(jq -r '.head.sha' <<< "${ORD_PR_DATA}")
+log::info "ORD_PR_COMMIT_HASH is: ${ORD_PR_COMMIT_HASH}"
+
+log::info "Fetch ORD service sources"
+cd /home/prow/ && git clone https://github.com/kyma-incubator/ord-service.git && cd ord-service && git checkout "${ORD_PR_COMMIT_HASH}" && cd ..
```
[the remark this answers is not in the document]
Comment (dzahariev) [reply prow/scripts/provision-vm-compass-ord.sh:53]: Not very easy, as the compass need to be in GOPATH, in order to be build and run. 
Let keep it that way.

**Review by NickyMateev — DISMISSED** [review]

File: templates/data/compass-smoke-test-data.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+templates:
+  - from: templates/generic.tmpl
+    render:
+      - to: ../prow/jobs/incubator/compass/compass-smoke-test.yaml
+        localSets:
+          jobConfig_default:
+            path_alias: github.com/kyma-incubator/compass
+            command: "/home/prow/go/src/github.com/kyma-project/test-infra/prow/scripts/provision-vm-compass-smoke-test.sh"
+            request_memory: 100Mi
+            request_mcpu: 50m
```
Comment (Halamix2) [inline templates/data/compass-smoke-test-data.yaml:10]:
```suggestion
            request_cpu: 50m
```

Comment (dzahariev) [reply templates/data/compass-smoke-test-data.yaml]: Done

**Review by Halamix2 — APPROVED** [review]

**Pull request merged**